frontend/src/app/admin/book-lists/book-list-form.tsx

```diff
@@ -0,0 +1,178 @@
+'use client';
+
+import { Button } from '@/components/ui/button';
+import { FormErrorMessage } from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
+import { BookList } from '@/services/book-lists';
+import { useTranslations } from 'next-intl';
+import { Controller, useForm } from "react-hook-form"
+import { zodResolver } from "@hookform/resolvers/zod"
+import { z } from "zod"
+import { AlertDialog, AlertDialogAction, AlertDialogCancel, AlertDialogContent, AlertDialogDescription, AlertDialogFooter, AlertDialogHeader, AlertDialogTitle, AlertDialogTrigger } from '@/components/ui/alert-dialog';
+import { useState } from 'react';
+
+interface BookListFormProps {
+  bookList?: BookList;
+  isSaving: boolean;
+  onOpenChange: (open: boolean) => void;
+  onSubmit: (data: BookList) => void;
+  onDelete: () => void;
+}
+
+export function BookListForm({ bookList, isSaving, onOpenChange, onSubmit, onDelete }: BookListFormProps) {
+  const t = useTranslations('Admin.BookLists');
+
+  const schema = z.object({
```

Comment:
Note: Zod objects are internally pretty big and it's a good practice to wrap the declaration of the object in a `useMemo` hook. And since you only use the `t()` function in it (and no other state), I think you can safely call useMemo() with no dependency.